SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMigration/SpatialTestCharacterMigration.cpp

```diff
@@ -75,90 +84,117 @@ void ASpatialTestCharacterMigration::PrepareTest()
 
 	// Move character forward
 	FSpatialFunctionalTestStepDefinition MoveForwardStepDefinition(/*bIsNativeDefinition*/ true);
-	MoveForwardStepDefinition.StepName = TEXT("Client1MoveForward");
-	MoveForwardStepDefinition.TimeLimit = 0.0f;
+	MoveForwardStepDefinition.StepName = TEXT("MoveForwardStepDefinition");
+	MoveForwardStepDefinition.TimeLimit = 5.0f;
 	MoveForwardStepDefinition.NativeTickEvent.BindLambda([this](float DeltaTime) {
-		AController* PlayerController = Cast<AController>(GetLocalFlowController()->GetOwner());
-		ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+		int32 Count = 0;
 
-		PlayerCharacter->AddMovementInput(FVector(1, 0, 0), 10.0f, true);
+		const UGameInstance* GameInstance = GetGameInstance();
+		const FString WorkerId = GameInstance->GetSpatialWorkerId();
 
-		bCharacterReachedDestination =
-			GetTargetDistanceOnLine(Origin, Destination, PlayerCharacter->GetActorLocation()) > -20.0f; // 20cm overlap
+		const USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(GetWorld()->GetNetDriver());
+		const bool bLoadBalanceStrategyValid = SpatialNetDriver != nullptr && SpatialNetDriver->LoadBalanceStrategy != nullptr;
 
-		if (bCharacterReachedDestination)
+		for (ATestMovementCharacter* Character : TActorRange<ATestMovementCharacter>(GetWorld()))
 		{
-			AssertTrue(bCharacterReachedDestination, TEXT("Player character has reached the destination on the autonomous proxy."));
-			FinishStep();
+			if (Character->IsLocallyControlled())
+			{
+				Character->AddMovementInput(FVector(1.0f, 0.0f, 0.0f), 10.0f, true);
+				AssertValue_Float(Character->Speed, EComparisonMethod::Less_Than, 650.0f, TEXT("Actor not speeding"));
+			}
+
+			bool bShouldRunRequires = true;
+			if (bLoadBalanceStrategyValid)
+			{
+				const VirtualWorkerId ExpectedVirtualWorker = 2;
+				const VirtualWorkerId LocalVirtualWorker = SpatialNetDriver->LoadBalanceStrategy->GetLocalVirtualWorkerId();
+				bShouldRunRequires = LocalVirtualWorker == ExpectedVirtualWorker;
+
+				if (bShouldRunRequires)
+				{
+					const VirtualWorkerId VirtualWorker = SpatialNetDriver->LoadBalanceStrategy->WhoShouldHaveAuthority(*Character);
+					RequireEqual_Int(VirtualWorker, ExpectedVirtualWorker, FString::Printf(TEXT("%s on %s crossed boundary"), *Character->GetName(), *WorkerId));
```

Comment:
Can't you run this on both servers? If you're running with no interest, you won't have a ATestMovementCharacter actor to operate on. And if you're running with interest, the call to `WhoShouldHaveAuthority` should still return the same value.